src/context.ts

```diff
@@ -69,9 +76,59 @@ function getContextFromWorkflow(): Context {
     }
   }
 
-  return context;
+  return {
+    commitDate: await getCommitDateFromWorkflow(context.sha, toolkit),
+    ...context
+  } as Context;
 }
 
 async function getContextFromGit(): Promise<Context> {
-  return await Git.context();
+  const ctx = await Git.context();
+
+  return {
+    commitDate: await Git.commitDate(ctx.sha),
+    ...ctx
+  } as Context;
+}
+
+async function getCommitDateFromWorkflow(sha: string, toolkit: Toolkit): Promise<Date> {
+  if (process.env.GITHUB_EVENT_PATH) {
+    const eventFile = await fs.readFile(process.env.GITHUB_EVENT_PATH, 'utf-8');
+    const event = JSON.parse(eventFile) as unknown as {
```

Comment:
you can just use `GitHub.context.payload` which carries the event payload similar to what is done in https://github.com/docker/metadata-action/blob/32323e53fe6fa1b9830e2eed4bd13fd97d948ed6/src/context.ts#L53

see this mock for more info: https://github.com/docker/metadata-action/blob/32323e53fe6fa1b9830e2eed4bd13fd97d948ed6/__mocks__/%40actions/github.ts#L10